experiments/large-result-sets/README.md

```diff
@@ -0,0 +1,50 @@
+# Performance gains for large result sets
+
+This experiment demonstrates the advantage of using [Query Builder's p.guessTotal](https://docs.adobe.com/help/en/experience-manager-65/developing/platform/query-builder/querybuilder-api.html) parameter to return a subset of Query Builder query results.
+
+## Problem
+
+By default, Query Builder reads the entire result set to determine the total number of hits for a query which can be costly in terms of both execution time and memory usage for queries that have large result sets. In this experiment, we'll attempt to show the performance benefit of using the `p.guessTotal` parameter to avoid calculating the exact size of the result set.
+
+## Setup
+
+AEM's QueryBuilder debugger tool will be used to execute [QueryBuilder-based] (https://docs.adobe.com/help/en/experience-manager-65/developing/platform/query-builder/querybuilder-api.html) queries on the JCR (Java Content Repository).
+
+## Test #1: `Calculate the exact total`
+
+1. Navigate to AEM's QueryBuilder debugger tool at: http://localhost:4502/libs/cq/search/content/querydebug.html.
+
+2. Find all content in the AEM system by typing the following text in the text area:
```

Comment:
"It is all content" is a phrase you'll hear.
I'd specify the idea by the folder.  Something like:
"Find all items in the /content folder in the JCR by..."